ipatests/pytest_ipa/integration/tasks.py

```diff
@@ -137,15 +137,22 @@ def wrapped(*args, **kwargs):
         return wrapped
     return wrapper
 
+
 def prepare_reverse_zone(host, ip):
     zone = get_reverse_zone_default(ip)
-    result = host.run_command(["ipa",
-                      "dnszone-add",
-                      zone], raiseonerr=False)
-    if result.returncode > 0:
-        logger.warning("%s", result.stderr_text)
+    option = host.run_command(["ipa",
+                               "dnszone-add",
+                               zone], raiseonerr=False)
+    if option.returncode > 0:
+        # Force DNSZone addition
+        cmd = ["ipa", "dnszone-add", "--skip-overlap-check", zone]
+        result = host.run_command(cmd, raiseonerr=False)
+        if result.returncode > 0:
+            logger.warning("%s", result.stderr_text)
+        return zone, result.returncode
     return zone, result.returncode
```

Comment:
In this codepath (option.returncode ==0), the variable `result` has not been assigned. I guess that you should return option.returncode instead.